crates/components/src/tilepicker.rs

```diff
@@ -145,8 +157,24 @@ impl Tilepicker {
             update_state.graphics.push_constants_supported(),
         );
 
+        let mut passages =
+            luminol_data::Table2::new(tilepicker_data.xsize(), tilepicker_data.ysize());
+        for (y, x) in (0..tilepicker_data.ysize()).cartesian_product(0..tilepicker_data.xsize()) {
+            passages[(x, y)] =
+                tileset.passages[tilepicker_data[(x, y, 0)].try_into().unwrap_or_default()];
```

Comment:
I'm getting a panic here:
```
The application panicked (crashed).
Message:  index out of bounds: the len is 1168 but the index is 1168
Location: crates\components\src\tilepicker.rs:164
```
`tileset.passages.xsize()` is 1168, `tilepicker_data.xsize() * tilepicker_data.ysize()` is 808.

This whole loop seems pretty inefficient anyway, `Table1` and `Table2` both expose `as_slice` methods is it possible you could use [slice.copy_from_slice](https://doc.rust-lang.org/stable/std/primitive.slice.html#method.copy_from_slice)?

Reply:
Looks like this is being caused by tiles and events on some maps having tile IDs larger than the tileset size. I'll have to add some code to the tile rendering as well to deal with this then.